pkg/scaling/scale_handler.go

```diff
@@ -620,3 +615,83 @@ func (h *scaleHandler) getScaledObjectState(ctx context.Context, scaledObject *k
 
 	return isScaledObjectActive, isScalerError, metricsRecord, nil
 }
+
+// / --------------------------------------------------------------------------- ///
+// / ----------             ScaledJob related methods               --------- ///
+// / --------------------------------------------------------------------------- ///
+
+// getScaledJobMetrics returns metrics for specified metric name for a ScaledJob identified by its name and namespace.
+// It could either query the metric value directly from the scaler or from a cache, that's being stored for the scaler.
+func (h *scaleHandler) getScaledJobMetrics(ctx context.Context, scaledJob *kedav1alpha1.ScaledJob) []scaledjob.ScalerMetrics {
+	cache, err := h.GetScalersCache(ctx, scaledJob)
+	if err != nil {
+		log.Error(err, "error getting scalers cache", "scaledJob.Namespace", scaledJob.Namespace, "scaledJob.Name", scaledJob.Name)
+		return nil
+	}
+	var scalersMetrics []scaledjob.ScalerMetrics
+	scalers, _ := cache.GetScalers()
+	for i, s := range scalers {
+		var queueLength float64
+		var targetAverageValue float64
+		isActive := false
+		maxValue := float64(0)
+		scalerType := fmt.Sprintf("%T:", s)
+
+		scalerLogger := log.WithValues("ScaledJob", scaledJob.Name, "Scaler", scalerType)
+
+		metricSpecs := s.GetMetricSpecForScaling(ctx)
+
+		// skip scaler that doesn't return any metric specs (usually External scaler with incorrect metadata)
+		// or skip cpu/memory resource scaler
+		if len(metricSpecs) < 1 || metricSpecs[0].External == nil {
+			continue
+		}
+
+		metrics, isTriggerActive, _, err := cache.GetMetricsAndActivityForScaler(ctx, i, metricSpecs[0].External.Metric.Name)
+		if err != nil {
+			scalerLogger.V(1).Info("Error getting scaler metrics and activity, but continue", "error", err)
+			cache.Recorder.Event(scaledJob, corev1.EventTypeWarning, eventreason.KEDAScalerFailed, err.Error())
+			continue
+		}
+
+		targetAverageValue = scaledjob.GetTargetAverageValue(metricSpecs)
+
+		var metricValue float64
+		for _, m := range metrics {
+			if m.MetricName == metricSpecs[0].External.Metric.Name {
+				metricValue = m.Value.AsApproximateFloat64()
+				queueLength += metricValue
+			}
+		}
+		scalerLogger.V(1).Info("Scaler Metric value", "isTriggerActive", isTriggerActive, metricSpecs[0].External.Metric.Name, queueLength, "targetAverageValue", targetAverageValue)
+
+		if isTriggerActive {
+			isActive = true
+		}
+
+		if targetAverageValue != 0 {
+			averageLength := queueLength / targetAverageValue
+			maxValue = scaledjob.GetMaxValue(averageLength, scaledJob.MaxReplicaCount())
+		}
```

Comment:
Sorry to be nit picky, but could you please also move this to the new package? To some a function that returns `queueLength` and `maxValue`? Something like `func calculateQueueLengthAndMaxValue(...) (queueLenght, maxValue float64)` ?

Reply:
The goal is to reduce the number of public functions within `scaledjob` package

Reply:
For more detail, I made **GetTargetAverageValue** a private. Please check.
Done.